alica_engine/src/engine/blackboard/KeyMapping.cpp

```diff
@@ -19,16 +19,26 @@ const KeyMapping::KeyMappingList& KeyMapping::getOutputMapping() const
     return _outputMapping;
 }
 
-void KeyMapping::addInputMapping(const std::string& parentKey, const std::string& childKey)
+void KeyMapping::addInputMapping(const std::variant<std::string, std::string>& parentKeyOrConstant, const std::string& childKey)
 {
-    assert(std::count(_inputMapping.begin(), _inputMapping.end(), std::make_pair(parentKey, childKey)) == 0);
-    _inputMapping.push_back(std::make_pair(parentKey, childKey));
+    assert(std::count_if(_inputMapping.begin(), _inputMapping.end(), [parentKeyOrConstant, childKey](Mapping& entry) {
```

Comment:
I think you check if the childKey is already mapped & if so just return (with a error log printed) instead of assert